packages/tailwindcss/src/utilities.ts

```diff
@@ -951,22 +951,39 @@ export function createUtilities(theme: Theme) {
     ['svh', '100svh'],
     ['lvh', '100lvh'],
     ['dvh', '100dvh'],
+    ['lh', '1lh'],
     ['min', 'min-content'],
     ['max', 'max-content'],
     ['fit', 'fit-content'],
   ]) {
-    staticUtility(`size-${key}`, [
-      ['--tw-sort', 'size'],
-      ['width', value],
-      ['height', value],
-    ])
-    staticUtility(`w-${key}`, [['width', value]])
-    staticUtility(`h-${key}`, [['height', value]])
-    staticUtility(`min-w-${key}`, [['min-width', value]])
-    staticUtility(`min-h-${key}`, [['min-height', value]])
-    if (key !== 'auto') {
-      staticUtility(`max-w-${key}`, [['max-width', value]])
-      staticUtility(`max-h-${key}`, [['max-height', value]])
+    let forWidth = key !== 'lh'
+    let forHeight = true
+    let forMax = key !== 'auto'
+
+    if (forHeight && forWidth) {
+      staticUtility(`size-${key}`, [
+        ['--tw-sort', 'size'],
+        ['width', value],
+        ['height', value],
+      ])
+    }
+
+    if (forWidth) {
+      staticUtility(`w-${key}`, [['width', value]])
+      staticUtility(`min-w-${key}`, [['min-width', value]])
+
+      if (forMax) {
+        staticUtility(`max-w-${key}`, [['max-width', value]])
+      }
+    }
+
+    if (forHeight) {
+      staticUtility(`h-${key}`, [['height', value]])
+      staticUtility(`min-h-${key}`, [['min-height', value]])
+
+      if (forMax) {
+        staticUtility(`max-h-${key}`, [['max-height', value]])
+      }
```

Comment:
I think it might be nicer to just store the common stuff in a variable and `concat` the extra stuff for the utilities that use them. Means a couple of loops instead of one but I think the simplicity/clarity of the code would be worth it.

Reply:
cleaned all that up 👍

Reply:
🤩